app/src/main/java/de/kabiskac/weekdaybug/ui/main/Fragment2.kt

```diff
@@ -22,7 +22,7 @@ class Fragment2 : Fragment() {
             inflater,
             R.layout.fragment2, container, false
         )
-        viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
+        viewModel = ViewModelProvider(requireActivity()).get(MainViewModel::class.java)
```

Comment:
Same as the comment above. This is using the same activity as the lifecycle owner in fragment2.